source/camera/CullingManager.cpp

```diff
@@ -0,0 +1,292 @@
+    mDebug(false),
+    mFirstIter(false),
+    mCm(cameraManager),
+    mCullCreaturesFlag(false),
+    mCullTilesFlag(false)
+{
+    mMyplanes =(Ogre::Plane(0, 0, 1, 0));
+    // init the Ogre vector
+    for (unsigned int i = 0; i < 8; ++i)
+    {
+        mOgreVectorsArray[i].x = static_cast<Ogre::Real>(0.0);
+        mOgreVectorsArray[i].y = static_cast<Ogre::Real>(0.0);
+        mOgreVectorsArray[i].z = static_cast<Ogre::Real>(0.0);
+    }
+}
+
+void CullingManager::cullTiles()
+{
+    mOldWalk = mWalk;
+    mWalk.mVertices.mMyArray.clear();
+    for (int ii = 0 ; ii < 8 ; ++ii)
+        mWalk.mVertices.mMyArray.push_back(Vector3i(mOgreVectorsArray[ii]));
+
+    // create a slope -- a set of left and rigth path
+    mWalk.convex_hull();
+    mWalk.buildSlopes();
+
+    if(mDebug)
+    {
+        LogManager::getSingleton().logMessage(LogMessageLevel::NORMAL,__FILE__,__LINE__ ,mOldWalk.debug());
+        LogManager::getSingleton().logMessage(LogMessageLevel::NORMAL,__FILE__,__LINE__ ,mWalk.debug());
+        
+    }
+
+    // reset index pointers to the begging of collections
+    mOldWalk.prepareWalk();
+    mWalk.prepareWalk();
+
+    newBashAndSplashTiles(SHOW | HIDE);
+}
+
+void CullingManager::startCreatureCulling()
+{
+    mCullCreaturesFlag = true;
+}
+
+void CullingManager::startTileCulling()
+{
+    getIntersectionPoints();
+
+    mWalk.mVertices.mMyArray.clear();
+    for (int ii = 0 ; ii < 8 ; ++ii)
+        mWalk.mVertices.mMyArray.push_back(Vector3i(mOgreVectorsArray[ii]));
+    mWalk.convex_hull();
+    mWalk.buildSlopes();
+    mOldWalk = mWalk;
+    mOldWalk.prepareWalk();
+    mWalk.prepareWalk();
+    hideAllTiles();
+    newBashAndSplashTiles(SHOW);
+
+    mCullTilesFlag = true;
+}
+
+void CullingManager::startDebugging()
+{
+    mDebug = true;
+}
+
+void CullingManager::stopCreatureCulling()
+{
+    mCullCreaturesFlag = false;
+}
+
+void CullingManager::stopTileCulling()
+{
+    mCullTilesFlag = false;
+    mOldWalk = mWalk;
+    mWalk.mVertices.mMyArray.clear();
+    for (int ii = 0 ; ii < 8 ; ++ii)
+        mWalk.mVertices.mMyArray.push_back(Vector3i(mOgreVectorsArray[ii]));
+
+    // create a slope -- a set of left and rigth path
+    mWalk.convex_hull();
+    mWalk.buildSlopes();
+
+    if(mDebug)
+    {
+        LogManager::getSingleton().logMessage(LogMessageLevel::NORMAL,__FILE__,__LINE__ ,mOldWalk.debug());
+        LogManager::getSingleton().logMessage(LogMessageLevel::NORMAL,__FILE__,__LINE__ ,mWalk.debug());
+        
+    }
+
+    // reset index pointers to the begging of collections
+    mOldWalk.prepareWalk();
+    mWalk.prepareWalk();
+
+    newBashAndSplashTiles(HIDE);
+    showAllTiles();
+
+}
+
+void CullingManager::hideAllTiles(void)
+{
+    GameMap* gm = mCm->mGameMap;
+    if (!gm)
+        return;
+
+    for (int jj = 0; jj < gm->getMapSizeY() ; ++jj)
+    {
+        for (int ii = 0; ii < gm->getMapSizeX(); ++ii)
+        {
+            gm->getTile(ii,jj)->hide();
```

Comment:
You were talking about coupling some days ago. Here, you are coupling GameEntity with the culling logic. That means that if one day, you somehow change something about culling, you may have to change the tile. From an object point of view, it is not good.

Concerning polymorphism (since it is the main argument you are pointing out):
1 - ATM, all entities are shown/hiden the same way. Even you seem to think that it is not to be overriden since the method is not virtual
2 - The show/hide logic is simple and not likely to change in the futur or for new game entities
3 - If one really wants to do some shadock show/hide feature, it can easily be changed

It is a choice to go one way or the other. But IMHO, the gain with low coupling is better than with polymorphism